indico/modules/events/registration/fields/base.py

```diff
@@ -98,12 +98,27 @@ def create_mm_field(self, registration=None):
         validators = self.get_validators(registration) or []
         if not isinstance(validators, list):
             validators = [validators]
-        if self.form_item.is_required and self.not_empty_if_required:
+
+        # Managers should be able to register without all required fields, but the personal data
+        # fields are mandatory either way.
+        from indico.modules.events.registration.models.form_fields import RegistrationFormPersonalDataField
+        required_personal_field = (
+            isinstance(self.form_item, RegistrationFormPersonalDataField) and
+            self.form_item.personal_data_type.is_required
+        )
+        skip_required = override_required and not required_personal_field
+
+        if self.form_item.is_required and self.not_empty_if_required and not skip_required:
             validators.append(not_empty)
+
+        mm_field_kwargs = self.mm_field_kwargs
+        if skip_required:
+            mm_field_kwargs['allow_none'] = True
+
         return self.mm_field_class(*self.mm_field_args,
-                                   required=self.form_item.is_required,
+                                   required=self.form_item.is_required and not skip_required,
```

Comment:
```suggestion
                                   required=(self.form_item.is_required and not skip_required),
```

Just for readability when kwarg values contain spaces